src/sentry/contexts.cpp

```diff
@@ -0,0 +1,340 @@
+		screen_data["dpi"] = DisplayServer::get_singleton()->screen_get_dpi(i);
+		screen_data["refresh_rate"] = Math::snapped(DisplayServer::get_singleton()->screen_get_refresh_rate(i), 0.01);
+		screen_data["position"] = DisplayServer::get_singleton()->screen_get_position(i);
+		screen_data["scale_factor"] = DisplayServer::get_singleton()->screen_get_scale(i);
+		screen_data["primary"] = i == DisplayServer::get_singleton()->get_primary_screen();
+
+		String orientation = _screen_orientation_as_string(i);
+		if (!orientation.is_empty()) {
+			screen_data["orientation"] = orientation;
+		}
+
+		screen_list.append(screen_data);
+	}
+
+	display_context["screens"] = screen_list;
+	return display_context;
+}
+
+Dictionary make_godot_engine_context() {
+	Dictionary godot_context = Dictionary();
+	ERR_FAIL_NULL_V(OS::get_singleton(), godot_context);
+	ERR_FAIL_NULL_V(Engine::get_singleton(), godot_context);
+
+	Dictionary version_info = Engine::get_singleton()->get_version_info();
+	godot_context["version"] = version_info.get("string", "");
+	godot_context["version_commit"] = version_info.get("hash", "");
+	godot_context["debug_build"] = OS::get_singleton()->is_debug_build();
+	godot_context["command_line_arguments"] = OS::get_singleton()->get_cmdline_args();
+	godot_context["mode"] = String(sentry::environment::get_environment());
+	godot_context["editor_build"] = OS::get_singleton()->has_feature("editor");
+
+	return godot_context;
+}
+
+Dictionary make_environment_context() {
+	Dictionary env_context = Dictionary();
+	ERR_FAIL_NULL_V(OS::get_singleton(), env_context);
+
+	env_context["sandboxed"] = OS::get_singleton()->is_sandboxed();
+	env_context["userfs_persistent"] = OS::get_singleton()->is_userfs_persistent();
+	env_context["granted_permissions"] = OS::get_singleton()->get_granted_permissions();
+
+#ifdef LINUX_ENABLED
+	String distribution_name = OS::get_singleton()->get_distribution_name();
+	if (!distribution_name.is_empty()) {
+		env_context["distribution_name"] = distribution_name;
+	}
+#endif
+
+	return env_context;
+}
+
+Dictionary make_performance_context() {
+	Dictionary perf_context = Dictionary();
+	ERR_FAIL_NULL_V(OS::get_singleton(), perf_context);
+	ERR_FAIL_NULL_V(Engine::get_singleton(), perf_context);
+	ERR_FAIL_NULL_V(Performance::get_singleton(), perf_context);
+	ERR_FAIL_NULL_V(RenderingServer::get_singleton(), perf_context);
+
+	if (!OS::get_singleton()) {
+		// Too early, bailing out.
+		return perf_context;
+	}
+
+	// * Injecting "Performance" context...
+
+	// Static memory allocation.
+	perf_context["static_memory_peak_usage"] = String::humanize_size(OS::get_singleton()->get_static_memory_peak_usage());
+	perf_context["static_memory_usage"] = String::humanize_size(OS::get_singleton()->get_static_memory_usage());
+
+	Dictionary meminfo = OS::get_singleton()->get_memory_info();
+	perf_context["thread_stack_size"] = String::humanize_size(meminfo["stack"]);
```

Comment:
It's the size of the current thread stack in bytes ("humanized" to avoid int32 overflow) as reported by https://docs.godotengine.org/en/stable/classes/class_os.html#class-os-method-get-memory-info
Since it's run on the main thread, it is about the main thread.

Reply:
Main worth adding `main_thread-stack_size` as a key then